lib/schedule/fetcher.ex

```diff
@@ -26,6 +26,20 @@ defmodule Schedule.Fetcher do
           files_source: files_source()
         }
 
+  @default_tables %{
+    routes: :routes,
+    route_patterns: :route_patterns,
+    timepoints_by_route: :timepoints_by_route,
+    timepoint_names_by_id: :timepoint_names_by_id,
+    shapes: :shapes,
+    stops: :stops,
+    trips: :trips,
+    blocks: :blocks,
+    calendar: :calendar,
+    runs: :run,
+    swings: :swings
+  }
```

Comment:
question: I don't understand what this name mapping is doing?

Reply:
So this part is a placeholder. Writing all of the data to the tables takes a long time - around 30+ seconds locally. It might be possible to do it table-by-table instead, but some of those (like trips, runs, and blocks) are big enough that they'll still probably take too long on their own, and then that leaves us with schedule data that may be inconsistent across tables. This basically means we'll need a procedure like the following:

1. There are two maps like this instead of one, one of which represents the current schedule data and the other of which is ready for `Schedule.Fetcher` to write to.
2. The `Schedule.Fetcher` process does its thing as it does now, but if new schedule data is found it writes it to the "standy" table names stored in the process state.
3. To hand this data off to the `Schedule` process, a `GenServer.call` is done as before. This replaces the current process state (now a mapping of table names) and returns the old map of tables.
4. The `Schedule.Fetcher` then takes this returned value, deletes the tables pointed to by those atoms, and stores it in its state for future use.

...and that's the main part of this that I haven't actually coded yet. I'll probably end up with a `@tables_a` and `@tables_b` (or maybe `@tables_red` and `@tables_blue` or something) instead.

Reply:
That makes sense. Before you go down that road, I would try to see if it's possible to do it within a transaction: you'd save yourself some infrastructure around the table name transitions if you can rely on Mnesia to handle it for you.

Reply:
Currently I'm having all the updates happen in the transaction, but I haven't yet tested how that works in practice.